cmd/gb-vendor/update.go

```diff
@@ -59,45 +59,38 @@ Flags:
 		}
 
 		for _, d := range dependencies {
-			url := d.Repository
-			p := d.Importpath
-
 			err = m.RemoveDependency(d)
 			if err != nil {
 				return fmt.Errorf("dependency could not be deleted from manifest: %v", err)
 			}
 
-			if err := os.RemoveAll(filepath.Join(ctx.Projectdir(), "vendor", "src", filepath.FromSlash(p))); err != nil {
+			if err := os.RemoveAll(filepath.Join(ctx.Projectdir(), "vendor", "src", filepath.FromSlash(d.Importpath))); err != nil {
 				// TODO(dfc) need to apply vendor.cleanpath here to remove indermediate directories.
 				return fmt.Errorf("dependency could not be deleted: %v", err)
 			}
 
-			repo, extra, err := vendor.DeduceRemoteRepo(p)
+			repo, _, err := vendor.DeduceRemoteRepo(d.Importpath)
 			if err != nil {
-				return fmt.Errorf("could not determine repository for import %q", p)
+				return fmt.Errorf("could not determine repository for import %q", d.Importpath)
 			}
 
-			wc, err := repo.Checkout("", "")
-			if err != nil {
-				return err
-			}
 
-			rev, err := wc.Revision()
+			wc, err := repo.Checkout(d.Branch, "")
 			if err != nil {
 				return err
 			}
 
-			branch, err := wc.Branch()
+			rev, err := wc.Revision()
 			if err != nil {
 				return err
 			}
 
 			dep := vendor.Dependency{
-				Importpath: p,
-				Repository: url,
+				Importpath: d.Importpath,
```

Comment:
I think we should be using the values from the fresh checkout here, rather than the original values from d. 

I think this because at best they are identical, in which case there is no semantic difference, but at worse, if they have changed, and they could, because DeduceRepo will get smarter over time, if we use the old values from d, then we won't be recording the updated state in the manifest.